app/controllers/resources_controller.rb

```diff
@@ -34,8 +34,76 @@ def push_to_onesky
     head :no_content
   end
 
+  def suggestions
+    tool_groups = []
+    ToolGroup.all.each { |tool_group| tool_groups << tool_group if match_params(tool_group, params) }
+
+    grouped_array = group_resources(tool_groups)
+    render json: order_results(grouped_array), status: :ok
+  end
+
   private
 
+  def order_results(grouped_array)
+    result = grouped_array.map do |key, values|
+      counter = values.size
+      sum = values.sum { |o| o[:tool_group_suggestions_weight] * o[:resource_tool_group_suggestions_weight] }
+      average = sum / counter
+      [Resource.find(key), average]
+    end
+
+    sorted_result = result.sort_by { |subarray| -subarray[1] }
+    sorted_result.map { |subarray| subarray[0] }
+  end
+
+  def group_resources(tool_groups)
+    result = tool_groups.flat_map do |tool_group|
+      tool_group.resource_tool_groups.map do |resource|
+        {
+          tool_group_suggestions_weight: tool_group.suggestions_weight,
+          resource_id: resource.resource_id,
+          resource_tool_group_suggestions_weight: resource.suggestions_weight
+        }
+      end
+    end
+
+    result.group_by { |hash| hash[:resource_id] }
+  end
+
+  def match_params(tool_group, params)
```

Comment:
I created a tool group with no defined rules and a single tool in it, but that tool is not being returned by the `/resources/suggestions` endpoint. Since the group had no rules it should be considered a match for any suggestion request.

This would probably also affect having only a single rule defined for a tool group, we will probably define a tool-group for a specific language only and not restrict it by country or the praxis scale, in which case that group should match as long as the specific language matches and not care about the rest of the defined filters

Reply:
or similarly there is a good chance we will define groups with just a country rule or just a praxis rule

Reply:
Ok, I think understood the idea. I'll work on tests for those cases and let you know when I've a case to test

Reply:
@frett I've added code for return as valid tool group when it has not rule/s

Reply:
The second part is ...

_"This would probably also affect having only a single rule defined for a tool group, we will probably define a tool-group for a specific language only and not restrict it by country or the praxis scale, in which case that group should match as long as the specific language matches and not care about the rest of the defined filters"_

@frett , so means that if a group tool has just one rule and it is about a "language rule" with a single language(not more than one language), this "tool group" should be considered as a match automatically without taking care of other param?